### DISCUSSION TITLE
How can I get the proper types for svgs?

### DISCUSSION THREAD

**Discussion by fuadsaud** [vercel/next.js#68985] [Help]
### Summary

By default, NextJS [defines the type of the default exports of SVG modules  as `any`](https://github.com/vercel/next.js/blob/canary/packages/next/image-types/global.d.ts#L10-L19).

I would like to override this behavior to avoid the linter from complaining about the usage of `any` typed values.

I have tried [a few proposed solutions](https://github.com/vercel/next.js/blob/canary/packages/next/image-types/global.d.ts#L10-L19) to no avail. I can see my project's type declarations are picked up by TS (as I see them in my editor when I try to navigate to the type definition of SVG values), but the types are still resolved as `any`.

Is this a limitation of TypeScript in that module declarations can't be completely overridden? If so, is there another known workaround?

Additionally, I'd be curious to understand why NextJS takes such an approach of defining SVGs modules as `any` (the comment about webpack/babel is not clear to me.

**Comment by Juneezee** [comment]
What I did is to use type assertion and cast the type to `StaticImageData`, for example:

```tsx
import Image, { type StaticImageData } from "next/image";
import logo from "../_assets/logo.svg";

    <Image
      src={logo as StaticImageData}
    />

```

**Reply by fuadsaud** [reply]
Right. What I'm trying to achieve is away to configure the module themselves to be typed as such, so no explicit casting is necessary.
